www/source/blog/2017-08-09-Habitat-plans-for-net-full-iis-apps.md

```diff
@@ -0,0 +1,301 @@
+
+Most of the `MSBUILD` targets, `.prop` files and the `msbuild.exe` needed to build .NET assblies can be found in your local `c:\windows\Microsoft.NET` folder. However the web targets are not. The easiest way to get those without installing Visual Studio is to use `nuget` again and install the `MSBuild.Microsoft.VisualStudio.Web.targets` package and then point `VSToolsPath` to its location.
+
+## Configuring IIS pools, sites and applications
+
+As mentioned earlier, we won't rely on Habitat for the installation of .NET or the right IIS features. However, it makes sense that Habitat be able to configure IIS in order that the application can run and behave as expected. We will leverage our run hook to do this.
+
+There are two approaches we can take in our hook. One is a bit dated and clunky but works everywhere. The other is more elegant but limited to machines where [WMF 5](https://www.microsoft.com/en-us/download/details.aspx?id=50395) (or higher) is installed.
+
+### appcmd.exe
+
+This is the old fashioned appcmd.exe of yore. For our purposes in our little sample app, it can get the job done:
+
+```powershell
+."$env:SystemRoot\System32\inetsrv\appcmd.exe" list apppool "{{cfg.app_pool}}" | Out-Null
+if($LASTEXITCODE -ne 0) {
+    Write-Host "Creating App Pool {{pkg.app_pool}}" -ForegroundColor Yellow
+    ."$env:SystemRoot\System32\inetsrv\appcmd.exe" add apppool /name:"{{cfg.app_pool}}"
+    ."$env:SystemRoot\System32\inetsrv\appcmd.exe" set apppool /apppool.name:"{{cfg.app_pool}}" /managedRuntimeVersion:v4.0
+}
+
+."$env:SystemRoot\System32\inetsrv\appcmd.exe" list site "{{cfg.site_name}}" | Out-Null
+if($LASTEXITCODE -ne 0) {
+    Write-Host "Creating Web Site {{cfg.site_name}}" -ForegroundColor Yellow
+    ."$env:SystemRoot\System32\inetsrv\appcmd.exe" add SITE /name:"{{cfg.site_name}}" /PhysicalPath:"$((Resolve-Path '{{pkg.svc_path}}').Path)" /bindings:http://*:{{cfg.port}}
+    Write-Host "Binding Web Site {{cfg.site_name}} to pool {{cfg.app_pool}}" -ForegroundColor Yellow
+    ."$env:SystemRoot\System32\inetsrv\appcmd.exe" set site /site.name:"{{cfg.site_name}}" "/[path='/'].applicationPool:{{cfg.app_pool}}"
+}
+
+."$env:SystemRoot\System32\inetsrv\appcmd.exe" list app /site.name:"{{cfg.site_name}}" | Findstr "{{cfg.app_name}}" | Out-Null
+if($LASTEXITCODE -ne 0) {
+    Write-Host "Creating Application {{cfg.app_name}} at path /{{pkg.name}} with physical path '$((Resolve-Path `"{{pkg.svc_var_path}}`").Path)'"
+    ."$env:SystemRoot\System32\inetsrv\appcmd.exe" add app /site.name:"{{cfg.site_name}}" /path:"/{{cfg.app_name}}" /PhysicalPath:"$((Resolve-Path '{{pkg.svc_var_path}}').Path)" /applicationPool:"{{cfg.app_pool}}"
+}
+
+."$env:SystemRoot\System32\inetsrv\appcmd.exe" start apppool "{{cfg.app_pool}}"
+."$env:SystemRoot\System32\inetsrv\appcmd.exe" start site "{{cfg.site_name}}"
+```
+
+This uses values in our configuration and creates the app pools, site and application folders needed to run our site.
```

Comment:
"This uses values in our configuration and creates the app pools, site and application folders needed to run our site." => "This uses values in our configuration to create the app pools and folders needed to run our site."